### PR TITLE
supporte quoted values for csv [approach 1]

### PR DESCRIPTION
**Pull request by leandromoh**
- [x] reader
- [ ] writer

### CODE REVIEW

**Comment by leandromoh** [comment]
note: add tests for fixed-length to check quotes only affects variable length

File: RecordParser/Engines/Reader/QuoteField.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+            scanned += unlook.IndexOf(singleQuote) + 1;
+            unlook = line.Slice(scanned);
+            position = 0;
+
+            while (true)
+            {
+                position += unlook.Slice(position).IndexOf(singleQuote);
+                position++;
+                if (unlook.Slice(position).IndexOf(singleQuote) == 0)
```
Comment (leandromoh) [inline RecordParser/Engines/Reader/QuoteField.cs]: I really dont want to spent time discovering the indexOf, just wanna check if it is zero
```suggestion
                if (unlook.Slice(position) is var temp 
                && temp.IsEmpty == false 
                && temp[0] == singleQuote)
```

**Comment by leandromoh** [comment]
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs
https://github.com/Cysharp/ZString/blob/a56ac542753eae6d956c4fc8aaf715cd9f680b79/src/ZString.Unity/Assets/Scripts/ZString/Utf16ValueStringBuilder.cs#L373

**Comment by yuzd** [comment]
need csv

**Comment by leandromoh** [comment]
> need csv

what do you mean @yuzd?

**Comment by yuzd** [comment]
Is csv feature already online?

**Comment by leandromoh** [comment]
> Is csv feature already online?

Yes, but quoted fields are not supported yet (this PR fix this). Take a look at classes `VariableLengthWriterBuilder` and/or `VariableLengthWriterSequentialBuilders`, there are examples of usage at [README](https://github.com/leandromoh/RecordParser#variable-length-reader) file as well in [unit tests](https://github.com/leandromoh/RecordParser/blob/master/RecordParser.Test/VariableLengthReaderBuilderTest.cs) and [benchmark](https://github.com/leandromoh/RecordParser/blob/master/RecordParser.Benchmark/VariableLengthReaderBenchmark.cs).

Feedbacks about your experience are welcome

**Comment by yuzd** [comment]
thanks

**Comment by leandromoh** [comment]
approach 1 was discontinued, decided to continue using approach 2 #32

**Pull request closed without merging**